tests/utils/test_msa_aligner.py

```diff
@@ -0,0 +1,145 @@
+
+        mafft = MAFFT(self.executable, self.tmpdir)
+        with self.assertRaises(ExecutionError):
+            mafft._run_aligner(args, env)
+
+        popen_mock.assert_called_once_with(args, stderr=subprocess.PIPE, encoding="utf-8", shell=True, env=env)
+        wait_mock.wait.assert_called_once_with()
+
+    @patch("shutil.which", return_value=1)
+    @patch.object(Path, Path.mkdir.__name__)
+    def test___create_new_sequences_file(self, *uninteresting_mocks):
+        directory = Path("tests/data/utils/create_new_sequences_file")
+        new_sequences = {"ACGT", "AAAA", "TTTT"}
+
+        mafft = MAFFT(self.executable, self.tmpdir)
+        mafft._create_new_sequences_file(directory, new_sequences)
+
+        self.assertTrue(filecmp.cmp(directory/"new_sequences.fa", directory/"new_sequences.truth.fa"))
+
+    @patch("shutil.which", return_value=1)
+    @patch.object(Path, Path.mkdir.__name__)
+    def test___get_aligner_name(self, *uninteresting_mocks):
+        mafft = MAFFT(self.executable, self.tmpdir)
+
+        expected = "MAFFT"
+        actual = mafft.get_aligner_name()
+
+        self.assertEqual(expected, actual)
+
+    @patch("shutil.which", return_value=1)
+    @patch.object(Path, Path.mkdir.__name__)
+    @patch("shutil.rmtree")
+    def test___cleanup_run(self, shutil_rmtree_mock, *uninteresting_mocks):
+        run_tmpdir = Path("run_tmpdir")
+
+        mafft = MAFFT(self.executable, self.tmpdir)
+        mafft._cleanup_run(run_tmpdir)
+
+        shutil_rmtree_mock.assert_called_once_with(run_tmpdir)
+
+    @patch("shutil.which", return_value=1)
+    @patch.object(MAFFT, MAFFT._create_new_sequences_file.__name__, return_value=Path("_create_new_sequences_file"))
+    @patch("os.environ", {"fake_key_1": "fake_value_1"})
+    @patch.object(Path, Path.mkdir.__name__)
+    @patch("make_prg.utils.msa_aligner.create_temp_dir", return_value=Path("fake_tmp_dir"))
+    @patch("builtins.open", new_callable=mock_open)
+    @patch.object(SeqIO, SeqIO.write.__name__)
+    @patch.object(MAFFT, MAFFT._run_aligner.__name__)
+    @patch("make_prg.utils.msa_aligner.load_alignment_file", return_value="updated_alignment")
+    @patch.object(MAFFT, MAFFT._cleanup_run.__name__)
+    def test___get_updated_alignment(self, cleanup_run_mock, load_alignment_file_mock, run_aligner_mock, write_mock,
+                                     open_mock, create_temp_dir_mock, *uninteresting_mocks):
+        mafft = MAFFT(self.executable, self.tmpdir)
+        alignment = make_alignment([
+            "AAAA",
+            "CCCC",
+            "GGGG",
+            "TTTT",
+        ])
+        new_seqs = {
+            "AATA",
+            "CCGC",
+            "GGCG",
+            "TTAT"
+
+        }
+
+        expected_updated_alignment = "updated_alignment"
+        actual_updated_alignment = mafft.get_updated_alignment(alignment, new_seqs)
+
+        self.assertEqual(expected_updated_alignment, actual_updated_alignment)
```

Comment:
what is actually being tested here? essentially that the execution of a fake mafft runs right?

Reply:
Answer in the response of the following comment